spec/requests/api/v3/repositories/revisions_resource_spec.rb

```diff
@@ -0,0 +1,129 @@
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+#++
+
+require 'spec_helper'
+require 'rack/test'
+
+describe 'API v3 Revisions resource', type: :request do
+  include Rack::Test::Methods
+  include Capybara::RSpecMatchers
+  include API::V3::Utilities::PathHelper
+
+  let(:revision) {
+    FactoryGirl.create(:changeset,
+                       repository: repository,
+                       comments: 'Some commit message',
+                       committer: 'foo bar <foo@example.org>'
+    )
+  }
+  let(:repository) {
+    FactoryGirl.create(:repository_subversion, project: project)
+  }
+  let(:project) {
+    FactoryGirl.create(:project, identifier: 'test_project', is_public: false)
+  }
+  let(:role) {
+    FactoryGirl.create(:role,
+                       permissions: [:view_changesets])
+  }
+  let(:current_user) {
+    FactoryGirl.create(:user, member_in_project: project, member_through_role: role)
+  }
+
+  let(:unauthorized_user) { FactoryGirl.create(:user) }
+
+  describe '#get' do
+    let(:get_path) { api_v3_paths.revision revision.id }
+    let(:expected_response) do
+      {
+        '_type' => 'Revision',
+        '_links' => {
+          'self' => {
+            'href' => "http://localhost:3000/api/v3/revisions/#{revision.id}"
+          },
+          'project' => {
+            'href' => "http://localhost:3000/api/v3/projects/#{project.id}",
+            'title' => project.title
+          },
+        },
+        'id' => revision.id,
+        'identifier' => revision.identifier,
+        'authorName' => revision.author,
+        'message' => {
+          'format' => 'plain',
+          'raw' => 'Some commit message',
+          'html' =>'<p>Some commit message</p>',
+        },
+        'createdAt' => revision.committed_on.utc.iso8601
+      }
+    end
+
+    context 'when acting as a user with permission to view work package' do
+      before(:each) do
+        allow(User).to receive(:current).and_return current_user
+        get get_path
+      end
+
+      it 'should respond with 200' do
+        expect(last_response.status).to eq(200)
+      end
+
+      describe 'response body' do
+        subject(:parsed_response) { JSON.parse(last_response.body) }
+
+        it 'should respond with revision in HAL+JSON format' do
+          expect(parsed_response['id']).to eq(revision.id)
```

Comment:
Just a hint:

You could also use the json-matchers (`be_json_eql`, etc) here. They give usually better error messages, e.g. "path `id` is missing" instead of "expected: 1, got: nil"